csharp/ql/test/utils/model-generator/theorems/TheoremSummaries.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Summaries;
+
+public class Theorems1<T> {
+    
+    public T Prop {
+        get { throw null; }
+        set { throw null; }
+    }
+
+    public Theorems1(T t)  { throw null; } 
+
+    public T Get() { throw null; }
+
+    public T Get(object x) { throw null; }
+
+    public T Id(T x) { throw null; }
+
+    public T Transform(int x, T y) { throw null; }
+
+    public S Transform<S>(S x) { throw null; }
+
+    public void Set(T x) { throw null; }
+
+    public void Set(int x, T y) { throw null; }
+
+    // No summary as S is unrelated to T
+    public void Set<S>(S x) { throw null; }
+
+    public int Apply(Func<T, int> f) { throw null; }
+}
+
+public class Theorems2 {
+
+    public T Transform<T>(T x) { throw null; }
+}
+
+public class CollectionTheorems1<T> : IEnumerable<T> {
```

Comment:
If it extends a known interface like this, which already has a known relation between its type variable and the way it's represented as contents, then we'd like the generated summaries to be compatible with the existing summaries. I.e. we'd like
```
var c = new CollectionTheorems1<String>();
c.Add(source());
sink(c.Last());
```
to have flow. Therefore we'll need the generated summary to just store in `Element` - not a freshly minted synthetic field.

Reply:
Yes, you are right!